PowerValidatedSolutions.psm1

```diff
@@ -50066,6 +50066,372 @@ Function Test-WMSubnetInput {
+    }
+
+    for ($createRandomCounter = 0; $createRandomCounter -lt 8; $createRandomCounter++) {
+        $randomString = -join (((48..57)+(65..90)+(97..122)) * 80 |Get-Random -Count 6 |%{[char]$_})
+        $cfgFilePath = Join-Path -path $certDir -ChildPath "$commonName-$randomString.cfg"
+        $orginalKeyFilePath = Join-Path -path $certDir -ChildPath "$commonName-$randomString.key"
+        if ((-Not(Test-Path -Path $cfgFilePath)) -and (-Not (Test-Path -Path $orginalKeyFilePath))) {
+			Write-Output "1" | Set-Content $cfgFilePath
+			Write-Output "1" | Set-Content $orginalKeyFilePath
+			if ((Test-Path -Path $cfgFilePath) -and (Test-Path -Path $orginalKeyFilePath)) {
+				Remove-Item $cfgFilePath
+				Remove-Item $orginalKeyFilePath
+				break
+			}
+        } else {
+			if($createRandomCounter -gt 6) {
+				Write-Error "Unable to create temporary files in directory $certDir." -ErrorAction Stop
+				Exit
+			}
+        }
+    }
+
+    $keyFilePath = Join-Path -path $certDir -ChildPath "$commonName.key"
+    $csrFilePath = Join-Path -path $certDir -ChildPath "$commonName.csr"
+    if (Test-Path -Path $keyFilePath) {
+        Write-Error "$keyFilePath file already exists.  Please remove or rename the current file or use an alternate directory path." -ErrorAction Stop
+    }
+    if (Test-Path -Path $csrFilePath) {
+        Write-Error "$csrFilePath file already exists.  Please remove or rename the current file or use an alternate directory path." -ErrorAction Stop
+    }
+    if ($expireDays -lt 2) {
+        Write-Error "Certificate expiration day ($expireDays) is less than 1 day.  Please provide a value greater than 1 day."
+    }
+
+    # Create temporary cfg file
+    $cfgString = '[ req ]
+    days = ' +$expireDays+ '
+    default_md = sha512
+    default_bits = ' +$keySize+ '
+    distinguished_name = req_distinguished_name
+    prompt = no
+    req_extensions = v3_req
+    
+    [ v3_req ]
+    basicConstraints = CA:FALSE
+    keyUsage = digitalSignature, keyEncipherment, dataEncipherment, nonRepudiation
+    extendedKeyUsage = serverAuth, clientAuth
+    subjectAltName = @alternate_names
+
+    [ req_distinguished_name ]
+    commonName = ' +$commonName+ '
+    countryName = ' +$country+ '
+    stateOrProvinceName = ' +$state+ '
+    localityName = ' +$locality+ '
+    0.organizationName = ' +$organization+ '
+    organizationalUnitName = ' +$organizationUnit+ '
+    
+    [ alternate_names ]
+    '
+    # Parsing SAN string
+    $sanArrary = $subjectAlternativeNames -split "\s*,\s*"
+    $sanCount = 1
+
+    foreach($san in $sanArrary) {
+        $cfgString = $cfgString + "DNS." + $sanCount + " = " + $san + "`n"
+        $sanCount++
+    }
+
+    # Generate RSA private key and CSR file
+    $cfgString | Out-File $cfgFilePath
+    openssl req -new -sha256 -nodes -out $csrFilePath -keyout $orginalKeyFilePath -config $cfgFilePath >> $null 2>$null
+    if ($opensslVersion -ge "3.0.0") {
```

Comment:
Question: Any reason we would not simply require OpenSSL 3.x or above?

OpenSSL 1.1 reached [end of life](https://www.openssl.org/blog/blog/2023/09/11/eol-111/index.html) on September 11, 2023 and there is no 2.x.

@bhumitra @GaryJBlake

Reply:
@garlicNova we should test with the latest openssl version and use it. Could you verify.

Reply:
I have implement a check for a version greater than 3.0.  
addressed in commit 9d50a853ffce55094985dea3d9e658670a7ab67a